providers/provider_default.go

```diff
@@ -33,8 +33,7 @@ var (
 )
 
 // GetLoginURL with typical oauth parameters
-func (p *ProviderData) GetLoginURL(redirectURI, state, _ string) string {
-	extraParams := url.Values{}
+func (p *ProviderData) GetLoginURL(redirectURI, state, _ string, extraParams url.Values) string {
```

Comment:
Why are we passing the extra params into all of these when it doesn't ever seem to be used?

Reply:
Currently all the per-provider implementations of `GetLoginURL` start by creating an empty `extraParams`.  Given we want to override some of the params based on what was passed to `/oauth2/start`, we need to construct the `extraParams` at the `oauthproxy` level and pass them down from there to the providers.

Reply:
Is the extraParams actually being used anywhere? I'm confused. When I was reviewing yesterday I didn't see any of the extraParams actually being used to go into the login URL, has this yet to be implemented?

Reply:
Yes, it gets passed on to `makeLoginURL` (in the next line) which applies the parameters to the generated URL.

Reply:
The logic to apply `extraParams` was already there in `makeLoginURL`, I just took out the special cases for prompt, approval_prompt and acr_values as those now fold into the new mechanism.

Reply:
Oh got it, that's what i've missed, thanks!